src/main/java/frc/robot/subsystems/AlgaeSubsystem.java

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import static frc.robot.Constants.AlgaeConstants.ALGAE_SLOT_CONFIGS;
+import static frc.robot.Constants.AlgaeConstants.DEVICE_ID_CANCODER;
+import static frc.robot.Constants.AlgaeConstants.DEVICE_ID_ROLLERMOTOR;
+import static frc.robot.Constants.AlgaeConstants.DEVICE_ID_WRISTMOTOR;
+import static frc.robot.Constants.AlgaeConstants.INTAKE_SPEED;
+import static frc.robot.Constants.AlgaeConstants.OUTTAKE_SPEED;
+import static frc.robot.Constants.AlgaeConstants.SCORE_SPEED;
+import static frc.robot.Constants.AlgaeConstants.WRIST_DOWN_POSITION;
+import static frc.robot.Constants.AlgaeConstants.WRIST_UP_POSITION;
+
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.HardwareLimitSwitchConfigs;
+import com.ctre.phoenix6.configs.Slot0Configs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VelocityTorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.ForwardLimitSourceValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+/**
+ * Algae subsystem does all jobs related to algae (intake, scoring, etc).
+ */
+public class AlgaeSubsystem extends SubsystemBase {
+
+  // define motors
+  public final TalonFX rollerMotor = new TalonFX(DEVICE_ID_ROLLERMOTOR);
+  public final TalonFX wristMotor = new TalonFX(DEVICE_ID_WRISTMOTOR);
+  // define how motors are controlled
+  private final VelocityTorqueCurrentFOC rollerControl = new VelocityTorqueCurrentFOC(0.0);
+  private final PositionVoltage wristControl = new PositionVoltage(0.0);
+  // define cancoder
+  public final static CANcoder cancoder = new CANcoder(DEVICE_ID_CANCODER);
+  // get variable for absolute position from cancoder
+  public final StatusSignal<Angle> absolutePosition = cancoder.getAbsolutePosition();
+
+  public AlgaeSubsystem() {
+    var rollerMotorConfig = new TalonFXConfiguration();
+    var wristMotorConfig = new TalonFXConfiguration();
+
+    rollerMotorConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+    wristMotorConfig.MotorOutput.NeutralMode = NeutralModeValue.Brake;
+
+    rollerMotorConfig.Slot0 = Slot0Configs.from(ALGAE_SLOT_CONFIGS);
+    wristMotorConfig.Slot0 = Slot0Configs.from(ALGAE_SLOT_CONFIGS);
+
+    // setting cancoder as a limit switch
+    HardwareLimitSwitchConfigs limitConfigs = new HardwareLimitSwitchConfigs();
+
+    limitConfigs.ForwardLimitSource = ForwardLimitSourceValue.RemoteCANcoder;
+    limitConfigs.ForwardLimitRemoteSensorID = cancoder.getDeviceID();
```

Comment:
Using RemoteCANCoder for a limit switch is actually a different thing, we want soft limits here